tests/cucumber/e2e/stepDefinitions/signup.Context.js

```diff
@@ -0,0 +1,31 @@
+const { Given, When, Then } = require ("@cucumber/cucumber")
+const { expect } = require("@playwright/test")
+const { SignUpPage } = require('../pageObjects/SignUpPage')
+const signUpPage = new SignUpPage
+
+Given('user has browsed into signup page', async function () {
+  await signUpPage.navigateSignUp()
+  
+  await expect(await signUpPage.navigateSignUp().getByText("SignUp")).toBeVisible()
+  });
+
+  When('user enters all the valid credentials:', async function (dataTable) {
+    const inputs = dataTable.hashes()
+
+    await page.fill(usernameSelector,inputs[0].username)
+    await page.type(emailInputSelector,inputs[0].Email)
+    if (inputs[0].Sex === "Male") {
+      await page.click(maleSelector)
+    } else {
+      await page.click(femaleSelector)
+    }
+    await page.fill(passwordInputSelector,inputs[0].Password)
+    await page.fill(passwordCheckSelector,inputs[0].ConfirmPassword)
+    await page.type(dobInputSelector,inputs[0].DateofBirth)
+    await page.click(signupButtonSelector)
+  });
+  
+  Then('user should see the success message {string}',async function (successMessage) {
+    expect(await page.locator(sinupVerificationSelector).innerText()).toBe(successMessage)
```

Comment:
You can create a function on your SignUpPage.js which returns the signup success message, for example `getSuccessMessage()` and use the function here to check the message.
```suggestion
    expect(await signUpPage.getSuccessMessage()).toBe(successMessage)
```